pkg/client/session.go

```diff
@@ -39,6 +46,11 @@ func (c *immuClient) OpenSession(ctx context.Context, user []byte, pass []byte,
 	}
 
 	c.ServiceClient = schema.NewImmuServiceClient(c.clientConn)
+	defer func(){
+		if err != nil {
+			_, _ = c.ServiceClient.CloseSession(ctx, new(empty.Empty))
```

Comment:
it's rare to defer the session closing before opening it. I'd expect the close statement after the line where it's opened

Reply:
Hi @jeroiraz here the logic that I apply as suggested (as I interpret it) by @byo : for each declared variable that can be reseted by an error, I add the condition that reset the relative variable just the next line after; so they are related.

An exception is `c.clientConn` that is outside scope and that's why I put it to the top!

Reply:
Yup, I think this particular `defer` should be a bit lower - right after `serviceClient.OpenSession` call.